### PR TITLE
Add ability for index managers to hide indicies

### LINKED ISSUE #14: An admin should be able to hide indices.
An admin should be able to hide indices.

### LINKED ISSUE #15: An admin should have to enter a reason for hiding  an index.
An admin should have to enter a reason for hiding  an index.

### PR DESCRIPTION
**Pull request by smacfarlane**
Closes #14 
Closes #15 

@gina-alaska/awrvi  This is not yet complete, but I need to take a step back from the details and make sure the workflow is what we're expecting.  Please review and comment.

### CODE REVIEW

File: app/views/indices/show.html.haml [hunk 1]
```diff
@@ -32,9 +32,11 @@
...
-      %li.list-group-item.active Manage index
-      - if can? :edit, @index
-        %li.list-group-item
-          = link_to 'Edit', edit_index_path(@index)
+  %ul.list-group
+    %li.list-group-item.active Manage index
+    - if can? :edit, @index
+      %li.list-group-item
+        = link_to 'Edit', edit_index_path(@index)
+    - if user_signed_in? && current_user.index_admin?
```
Comment (teknofire) [inline app/views/indices/show.html.haml]: Should see about adding a custom permission to the ability file to handle this

File: app/models/ability.rb [hunk 2]
```diff
@@ -14,10 +14,14 @@ def initialize(user)
 
     unless user.new_record?
       can :create, Index
-      can :manage, Index, user_id: user.id
+      can :manage, Index, user_id: user.id, hidden: false
+    end
+
+    if user.index_admin? && namespace == 'manage'
+      can [:edit, :update], Index
```
Comment (teknofire) [inline app/models/ability.rb]: `:update` will catch the `:edit` action as well

probably need to see about splitting this up a bit to handle the different place we have to worry about permissions being checked from.

1) Normal index show, which won't be in the manage namespace - `can :hide, Index if user.index_admin?`
2) Manage index update, which will have the manage namespace - `can :update, Index`

**Comment by grimm** [comment]
:+1:

**Comment by teknofire** [comment]
Looks like there might be a permission redirect loop problem that we need to clean up.

If I hide a index then try to go and click the link to view it I get stuck in a redirect loop, although it's not something specific to this PR, I think it's a general permission denied handling problem.

**Comment by teknofire** [comment]
I'm also not going to worry about trying to clean up the styling for the link and form as the other PR's make some heavy changes to the views that those would integrate with.

I think at this point having a separate pull request that fixes the styling is fine.

:+1:

**Pull request merged**